text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1067 @@
+  repeated Stacktrace stacktraces = 7;
+
+  // A lookup table of Mappings. Other messages refer to Mappings in this table by index.
+  repeated Mapping mappings = 8;
+
+  // A lookup table of Locations. Other messages refer to Locations in this table by index.
+  repeated Location locations = 9;
+
+  // A lookup table of Functions. Other messages refer to Functions in this table by index.
+  repeated Function functions = 10;
+
+  // A lookup table of Links to trace spans associated with this profile. Other messages refer to Links in this table by index. The first message must be an empty Link — this represents a null Link.
+  repeated Link links = 11;
+
+  // A lookup table of AttributeSets. Other messages refer to AttributeSets in this table by index. The first message must be an empty AttributeSet — this represents a null AttributeSet.
+  repeated AttributeSet attribute_sets = 12;
+
+  // A lookup table of strings. Other messages refer to strings in this table by index.
+  // The 0-th element must be an empty string ("").
+  repeated string string_table = 13;
+
+  // List of profile types included in this profile. The first item in the list is considered to be the "default" profile type. Example profile types are allocated objects or allocated bytes.
+  repeated ProfileType profile_types = 14;
+}
+
+// Represents a relationship between a Sample (ephemeral structure represented by references to a Stacktrace, AttributeSet, Link + value and a timestamp) and a trace span. This allows for linking between specific Samples within a profile and traces.
+message Link {
+  // A unique identifier of a trace that this linked span is part of. The ID is a
+  // 16-byte array.
+  bytes trace_id = 1;
+  // A unique identifier for the linked span. The ID is an 8-byte array.
+  bytes span_id = 2;
+}
+
+// AttributeSet represents a set of attributes. Multiple Samples, Locations and Mappings may have the same attributes and that's why this is a separate message. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+message AttributeSet {
+  // Attributes associated with a specific Sample, Location or a Mapping.
+  // attributes is a collection of key/value pairs. Note, global attributes
+  // like server name can be set using the resource API. Examples of attributes:
+  //
+  //     "/http/user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36"
+  //     "/http/server_latency": 300
+  //     "abc.com/myattribute": true
+  //     "abc.com/score": 10.239
+  //
+  // The OpenTelemetry API specification further restricts the allowed value types:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute
+  // Attribute keys MUST be unique (it is not allowed to have more than one
+  // attribute with the same key).
+  repeated opentelemetry.proto.common.v1.KeyValue attributes = 1;
```

Comment:
Not saying we should do this, but would there be value to a canonical ordering here? E.g. sorted by key?